product_identification/models/productid_number.py

```diff
@@ -0,0 +1,50 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductIdNumber(models.Model):
+    _name = "product.product.id_number"
+    _description = "Product Identification Number"
+
+    name = fields.Char(string="ID Number", required=True)
+    category_id = fields.Many2one(
+        "product.product.id_category", string="Category", required=True
+    )
+    issued_by = fields.Many2one("res.partner")
+    date_issued = fields.Date()
+    expiry_date = fields.Date()
+    place_of_issue = fields.Char(string="Place of Issue")
+    product_id = fields.Many2one("product.product", string="Product")
```

Comment:
```suggestion
    product_id = fields.Many2one("product.product", string="Product", required=True)
```
I guess it doesn't make sense to have an inentification without product. Or did i missed something ?